### ISSUE TITLE
News-menu conflict with the JustLight and Primer themes when switching from light to black palette

### ISSUE THREAD

**Issue by gudjonsi** [Tywed/news-menu#17]
I'm not sure whether this is a theme issue (i.e. JustLight and Primer) or an issue with the news-menu module. This is not a problem with the standard webtrees themes. When using either the JustLight or Primer theme with the news-menu module and changing from light to black palettes of the themes the text in the ShowNews (module/_news-menu_/ShowNews) is white on white background and therefore not readable. In the news overview (module/_news-menu_/Page) or in the EditNews (module/_news-menu_/EditNews/) the light and black palettes of the themes do not cause this problem, only in ShowNews, see enclosed images for the JustLight theme. I hope this explanation  makes some sense in describing the problem. The News-Menu module seems to take into account the JustLight and Primer themes css but not when showing a specific news in the Show-News mode.
Regards.

![news-menu-justlight-theme-1](https://github.com/user-attachments/assets/aae94260-6a79-4e84-a695-d11db6c7a239)
![news-menu-justlight-theme-2](https://github.com/user-attachments/assets/617da1d7-1805-438c-b156-1fc4697c22d3)
![news-menu-justlight-theme-3](https://github.com/user-attachments/assets/ac8210cc-cc7d-4194-a793-4886946b9f20)
![news-menu-justlight-theme-4](https://github.com/user-attachments/assets/b2930879-4be5-4630-8cd5-5268d738619b)
![news-menu-justlight-theme-5](https://github.com/user-attachments/assets/5430041b-1c63-475b-b1e0-cb29405cd101)
![news-menu-justlight-theme-6](https://github.com/user-attachments/assets/b836d724-3bf0-44a8-b4c4-1a4098c46ae2)
![news-menu-justlight-theme-7](https://github.com/user-attachments/assets/a5399e3a-410f-4f2b-acc4-458c53080a9a)
![news-menu-justlight-theme-8](https://github.com/user-attachments/assets/848cd054-62ca-4c6f-8fde-8018bf363a46)

**Comment by Tywed** [comment]
Yes, you are right, the module has not been tested with all third-party themes, only with standard ones. Since the show.phtml file uses non-standard css for the block (<div class="card bg-light">), it can be replaced in two places with <div class="card mb-4 wt-block wt-block-gedcom_news">. Then the problem should go away.

**Comment by gudjonsi** [comment]
Thanks, after some trial and error I was able to make the necessary changes to the show.phtml so now the module works with both the JustLight Theme and the Primer-Theme including the light and black palettes of those themes. I've also tested these changes on the standard webtrees themes and they work as well without any problems. Also the Argon-Light Theme and Rural Theme were tested without any problems. With these changes the News-Menu module takes into account the css of the different themes and is consequently more coherent with the selected themes for webtrees.

Following changes were done to make the show.phtml work for different third party themes including the standard webtrees themes. The changes also include padding around the image and rounding of the image corners to make it better fit and align within the wt-block of the show.phtml:

```
Line 24 (original):    <div class="card bg-light ">
Line 24 (changes: replace bg-light with wt-block):     <div class="card wt-block">

Line 34 (original):    <?php if (isset($news_media)) {
Line 35 (original):    echo $news_media->displayImage(1000, 667, 'contain', ['class' => 'w-100 me-3']);
Line 36 (original):     }
Line 37 (original):    ?>

Line 34 (changes: none):    <?php if (isset($news_media)) {
Line 35 (changes: add p-4 and rounded-5 within class):    echo $news_media->displayImage(1000, 667, 'contain', ['class' => 'p-4 rounded-5 w-100 me-3']);
Line 36 (changes: none):    }
Line 37 (changes: none):    ?>

Line 213 (original):  <div class="card bg-light text-dark">
Line 213 (changes: replace bg-light and text-dark with wt-block):   <div class="card wt-block">

Line 228 (original):  <a href="<?php echo e(route('module', ['module' => $module_name, 'action' => 'ShowNews', 'news_id' => $article->news_id, 'tree' => $tree->name()])) ?>" class="text-dark">
Line 228 (changes: delete: class="text-dark"):   <a href="<?php echo e(route('module', ['module' => $module_name, 'action' => 'ShowNews', 'news_id' => $article->news_id, 'tree' => $tree->name()])) ?>">
```

**Comment by gudjonsi** [comment]
I also changed the edit-phtml to show the typical block visible in webtrees themes with some padding to make it more coherent with the different third party themes as well the standard webtrees themes.
```

Line 21 (original):    <div class="col-md-8 mx-auto">
Line 21 (changes: add card wt-block and p-4 within the class):    <div class="card wt-block p-4 col-md-8 mx-auto">

```

**Comment by gudjonsi** [comment]
Perhaps the above changes to show.phtml and edit.phtml could be incorporated in the next release of News-Menu Module, i.e. if you agree with these changes. Thanks again for a great webtrees module.
